extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/metrics/DoubleGauge.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.metrics;
+
+import io.opentelemetry.api.common.Attributes;
+import javax.annotation.concurrent.ThreadSafe;
+
+/** A gauge instrument that synchronously records {@code double} values. */
+@ThreadSafe
+public interface DoubleGauge {
+  /**
+   * Set the gauge value.
+   *
+   * @param value The current gauge value.
+   */
+  void set(double value);
+
+  /**
+   * Records a value with a set of attributes.
+   *
+   * @param value The current gauge value.
+   * @param attributes A set of attributes to associate with the value.
+   */
+  void set(double value, Attributes attributes);
+
+  // TODO(jack-berg): should we add overload with Context argument?
```

Comment:
Yes.

Reply:
I'm going to open an issue for this and solve in a separate PR. The synchronous use cases I'm aware of require the instrumentation to subscribe to notifications about change in some value, rather than fetching the current value which caters to async gauge. Not clear when context would be available in one of these notification event streams. I'm open to it being a use case, but its not as clear cut as what's in this PR.